analytics-core/src/main/java/com/segment/analytics/SegmentDispatcher.java

```diff
@@ -251,7 +251,12 @@ void performFlush() {
     try {
       int payloadsUploaded = networkExecutor.submit(new Callable<Integer>() {
         @Override public Integer call() throws Exception {
-          return upload();
+            try {
+                return upload();
+            } catch(IOException ioe) {
+                error(ioe, "Upload failed: " + ioe.getMessage());
+                return -1;
```

Comment:
No reason - I had to return a number so I just figured on Exception return a negative number.  I changed it to zero locally and I can still repro the error.